packages/next-app/src/hooks/useConfirmations.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from "react";
+import { useBlockNumber, useWaitForTransaction } from "wagmi";
+
+type BlockConfirmations = number | undefined;
+
+const useConfirmations = (latestBlockHash: string | undefined): number | undefined => {
+  const [{ data }] = useBlockNumber();
```

Comment:
```suggestion
  const [{ data }] = useBlockNumber({ watch: true });
```

I'd suggest adding "watch" here, so that the confirmations number increases whenever a new block gets mined.